### DISCUSSION TITLE
Эффекты и будильник в приложении

### DISCUSSION THREAD

**Discussion by Cyxapuk11** [vvip-68/GyverPanelWiFi#298] [Задавайте вопросы]
Есть несколько проблем (багов) и трудностей, с которыми столкнулся в приложении. 
1) При попытке выбрать текущий эффект в соотв. вкладке, эффекты начинают быстро переключаться по списку в сторону "часы". На часах останавливается и воспроизводятся только они. Если выбрать режим "демо" на вкладке "матрица", то эффекты воспроизводятся поочередно в произвольном порядке с заданным интервалом (крч как и надо). Пробовал менять эти настройки, результата не добился:
![image](https://github.com/vvip-68/GyverPanelWiFi/assets/155484990/b2121694-9665-48ab-888c-ec6363c6e00c)

2) У будильника в приложении сбиваются настройки длительности и эффекта. Кроме того, когда выбираешь эффект и жмешь "установить", автоматом выбирается вышестоящий другой эффект будильника (но применяется выбранный пользователем).  
И мне не совсем понятны настройки длительности. Будильник начинается раньше установленного времени на "длит рассвета+длит будильника", но на протяжении длительности рассвета ничего не происходит. Потом на всю длительность будильника включается максимально ярко выбранный эффект. 
Думаю яркость должна нарастать плавно, или по крайней мере нам не нужно две настройки времени. Скорее всего это тоже какой-то баг.

Ткните если такую тему уже поднимали, я не нашел подобного

**Comment by vvip-68** [comment]
А ни у кого ничего подобного и не было. Вы первый и единственный, похоже.
Вряд ли тут кто-то сможет дать вам ответ. Я не знаю почему такое может происходить.

1. Эти настройки вообще никакого отношения к эффектам не имеют. Это только для роликов с SD-карты
2. Длительность - за указанное время ДО будильника должен включаться выбранный эффект с плавно нарастающей яркостью (длительность рассвета). В назначенное время яркость должна включаться на максимальную, и кажется, там просто белый яркий свет на указанное время (длительность будильника). То что описываете вы - никакому логическому объяснению не поддается и причин я даже придумать не могу, чтобы оно так себя вело.


Ждите, может кто-нибудь ответит. Хотя, вряд ли. Никто ранее о таком не писал - значит, скорее всего ни у кого такое не наблюдалось.

**Reply by shkolin1973** [reply]
По будильнику у меня следующий глюк:
1. Длительность рассвета (по умолчанию 10мин) - отрабатывает нормально. Яркость плавно нарастает в течении 10мин до назначенного времени.
2.  Длительность будильника (по умолчанию 1мин) - не возможно изменить это время. Меняешь допустим на 2мин, записываешь. Вышел из приложения, зашёл. Всё, опять стоит 1мин.

**Reply by vvip-68** [reply]
Исправил. Забирайте обновленную версию из хранилища.

**Reply by shkolin1973** [reply]
Спасибо! Буду пробовать.

**Reply by shkolin1973** [reply]
Теперь всё работает правильно. Спасибо!
